controllers/osbuildenvconfig_controller_test.go

```diff
@@ -0,0 +1,740 @@
+	"github.com/project-flotta/osbuild-operator/internal/repository/secret"
+	"github.com/project-flotta/osbuild-operator/internal/repository/service"
+
+	"github.com/project-flotta/osbuild-operator/controllers"
+)
+
+var _ = Describe("OSBuildEnvConfig Controller", func() {
+	const (
+		operatorNamespace        = "osbuild"
+		caIssuerName             = "osbuild-issuer"
+		dbSecretName             = "composer-db"
+		instanceName             = "env"
+		osBuildOperatorFinalizer = "osbuilder.project-flotta.io/osBuildOperatorFinalizer"
+		composerCertificateName  = "composer-cert"
+		templatesDir             = "../resources/templates"
+	)
+
+	var (
+		mockCtrl *gomock.Controller
+
+		scheme *runtime.Scheme
+
+		osBuildEnvConfigRepository *osbuildenvconfig.MockRepository
+		certificateRepository      *certificate.MockRepository
+		configMapRepository        *configmap.MockRepository
+		deploymentRepository       *deployment.MockRepository
+		serviceRepository          *service.MockRepository
+		secretRepository           *secret.MockRepository
+
+		reconciler     *controllers.OSBuildEnvConfigReconciler
+		requestContext context.Context
+
+		errNotFound error
+		errFailed   error
+
+		instance osbuildv1alpha1.OSBuildEnvConfig
+
+		request = ctrl.Request{
+			NamespacedName: types.NamespacedName{
+				Name: instanceName,
+			},
+		}
+
+		resultRequeue = ctrl.Result{Requeue: true}
+		resultDone    = ctrl.Result{}
+
+		composerCertificateSecret = corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: operatorNamespace,
+				Name:      composerCertificateName,
+			},
+		}
+	)
+
+	BeforeEach(func() {
+		os.Setenv("WORKING_NAMESPACE", operatorNamespace)
+		os.Setenv("CA_ISSUER_NAME", caIssuerName)
+		os.Setenv("TEMPLATES_DIR", templatesDir)
+		err := conf.Load()
+		Expect(err).To(BeNil())
+
+		mockCtrl = gomock.NewController(GinkgoT())
+
+		osBuildEnvConfigRepository = osbuildenvconfig.NewMockRepository(mockCtrl)
+		certificateRepository = certificate.NewMockRepository(mockCtrl)
+		configMapRepository = configmap.NewMockRepository(mockCtrl)
+		deploymentRepository = deployment.NewMockRepository(mockCtrl)
+		serviceRepository = service.NewMockRepository(mockCtrl)
+		secretRepository = secret.NewMockRepository(mockCtrl)
+
+		scheme = runtime.NewScheme()
+		err = clientgoscheme.AddToScheme(scheme)
+		Expect(err).To(BeNil())
+		err = osbuildv1alpha1.AddToScheme(scheme)
+		Expect(err).To(BeNil())
+		err = certmanagerv1.AddToScheme(scheme)
+		Expect(err).To(BeNil())
+
+		reconciler = &controllers.OSBuildEnvConfigReconciler{
+			Scheme:                     scheme,
+			OSBuildEnvConfigRepository: osBuildEnvConfigRepository,
+			CertificateRepository:      certificateRepository,
+			ConfigMapRepository:        configMapRepository,
+			DeploymentRepository:       deploymentRepository,
+			ServiceRepository:          serviceRepository,
+			SecretRepository:           secretRepository,
+		}
+
+		requestContext = context.TODO()
```

Comment:
you can use `request.Context()`

Reply:
Are you sure? I think you're confusing with another type of `request`

Reply:
by a mistake I thought `request` is from type "http.Request" 
So you can't use it like that